### PR TITLE
require email and two linked accounts to request an organization

### PR DESCRIPTION
**Pull request by NatalieEO**
https://ucsc-cgl.atlassian.net/browse/SEAB-3275

### CODE REVIEW

File: src/app/organizations/registerOrganization/requireAccountsModal/require-accounts-modal.component.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<h1 mat-dialog-title>User Verification</h1>
+<p>
+  For verification purposes, you are required to have at least two external accounts linked to your Dockstore account. However, it's
+  preferrable to have more than two because Dockstore curators will use that information to verify that your organization request is
+  legitimate.
+</p>
+<p>You have {{ numLinkedAccounts }} account(s) linked.</p>
+<div mat-dialog-actions class="pull-right">
```
Comment (garyluu) [inline src/app/organizations/registerOrganization/requireAccountsModal/require-accounts-modal.component.html:8]: `Dialog and card action buttons aligned right using fxLayout="row" fxLayoutAlign="end center" fxLayoutGap="1rem"` from https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide

File: src/app/organizations/registerOrganization/requireAccountsModal/require-accounts-modal.component.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  selector: 'app-require-accounts-modal',
+  templateUrl: './require-accounts-modal.component.html',
+})
+export class RequireAccountsModalComponent implements OnInit {
+  public numLinkedAccounts: number = 0;
+  constructor(private tokenQuery: TokenQuery, private matDialog: MatDialog) {}
+
+  ngOnInit(): void {
+    this.tokenQuery.tokens$.subscribe((tokens: TokenUser[]) => {
+      tokens.forEach((token) => {
```
Comment (coverbeck) [inline src/app/organizations/registerOrganization/requireAccountsModal/require-accounts-modal.component.ts:18]: Can't you just do `this.numLinkedAccounts = tokens.length - 1`?

File: src/app/organizations/registerOrganization/requireAccountsModal/require-accounts-modal.component.html [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<h1 mat-dialog-title>User Verification</h1>
+<p>
+  For verification purposes, you are required to have at least two external accounts linked to your Dockstore account. However, it's
+  preferrable to have more than two because Dockstore curators will use that information to verify that your organization request is
+  legitimate.
+</p>
+<p>You have {{ numLinkedAccounts }} account(s) linked.</p>
```
Comment (coverbeck) [inline src/app/organizations/registerOrganization/requireAccountsModal/require-accounts-modal.component.html:7]: I'm a fan of figuring out whether you need the parens or not. :) One of my pet peeves in DOS (and maybe still Windows?).

I think Angular has a pipe for this.

**Review by coverbeck — COMMENTED** [review]
Can you please add a screenshot of what this looks like with the error?

File: src/app/organizations/registerOrganization/register-organization.component.html [hunk 4]
```diff
@@ -77,6 +78,7 @@ <h1 mat-dialog-title>{{ title }}</h1>
     [disabled]="!registerOrganizationForm.valid"
     (click)="createOrUpdateOrganization()"
     cdkFocusInitial
+    mat-dialog-close
```
[the remark this answers is not in the document]
Comment (NatalieEO) [reply src/app/organizations/registerOrganization/register-organization.component.html:81]: Ohh yes, you're right it does close. The error is displayed at the top of the pag
Screen Shot 2021-09-21 at 1 14 46 PM
e

[the remark this answers is not in the document]
Comment (garyluu) [reply src/app/organizations/registerOrganization/register-organization.component.html:81]: i don't think we've settled on what the behavior should be. but i think i we'd generally want to keep the modal open so they can edit the form (like change the name). Up to others to decide though.

[the remark this answers is not in the document]
Comment (NatalieEO) [reply src/app/organizations/registerOrganization/register-organization.component.html]: I ended up removing the mat-dialog-close, so it's back to staying open

File: src/app/organizations/registerOrganization/requireAccountsModal/require-accounts-modal.component.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+
+@Component({
+  selector: 'app-require-accounts-modal',
+  templateUrl: './require-accounts-modal.component.html',
+})
+export class RequireAccountsModalComponent implements OnInit {
+  public numLinkedAccounts: number = 0;
+  private ngUnsubscribe: Subject<{}> = new Subject();
```
Comment (garyluu) [inline src/app/organizations/registerOrganization/requireAccountsModal/require-accounts-modal.component.ts:16]: this unsubscribe needs to be linked to the onDestroy method or something else in order for it to unsubscribe. my recommendation is still to pipe map so...

```
numLinkedAccounts$ = this.tokenQuery.tokens$.pipe(map(tokens => tokens.length - 1));
```

then you would have to async pipe in the html like:

`numLinkedAccounts$ | async`

**Review by garyluu — APPROVED** [review]
LGTM doesn't look related

**Comment by coverbeck** [comment]
> Can you please add a screenshot of what this looks like with the error?

Sorry, I meant the validation error (when you don't have at least 2 accounts) -- not the error when a call to the web service fails.

**Comment by NatalieEO** [comment]
> > Can you please add a screenshot of what this looks like with the error?
> 
> Sorry, I meant the validation error (when you don't have at least 2 accounts) -- not the error when a call to the web service fails.
Screen Shot 2021-09-22 at 3 09 50 PM


Ah, there isn't an error displayed. The "Next" button is disabled.

**Review by denis-yuen — APPROVED** [review]

**Review by coverbeck — APPROVED** [review]
> Ah, there isn't an error displayed. The "Next" button is disabled.

I was thinking of the `mat-error` snippets I'd seen, but yes, now I realize it's just a new dialog.

**Pull request merged**